File: Magic/history.py
import datetime, os, webbrowser
from pathlib import Path

currentuser = ''


def user_file(command: str = '', task_did: str = '',username = None, ) -> None:
    """[Used to save the history of the user]"""
    maxlength = 100
    username = currentuser if username is None else username
    with open(f"{os.getcwd()}/resources/ {username}.elsa", "a") as history:
        history.write(f"{'-' * maxlength}\n DATE{datetime.datetime.now()} USER INPUT: {command} OUTPUT: {task_did}\n{'-' * maxlength}\n\n")


def user_read(event = "", username: str = "dummy") -> None:
    """[Open the user history file]"""
    if not Path(userpth := f"{os.getcwd()}/resources/ {username}.elsa").exists():
        userpth = f'{os.getcwd()}/resources/ dummy.elsa'
    webbrowser.open(userpth)


def clear_history(name: str) -> None:
    """[Clears the history of the user]"""
    with open(f"{os.getcwd()}/resources/ {name}.elsa", "w") as history: history.write("")

File: Magic/History.cs
using HistoryGUI;
namespace Magic;

public class History

{
    private readonly string _username;
    private readonly DataFileManager _dataFileManager;

    public History(string username)
    {
        _username = username;
        _dataFileManager = new DataFileManager(_username);
    }
    public void Write(string userInput,string output)
    { 
        //This file is to save the history to user file
        
        _dataFileManager.Write("History",$"{DateTime.Now:F} : {userInput} : {output}");
    }

    public void Read()
    {
        string st = string.Join("\r\n", _dataFileManager.GetHistory());
        HistoryDisplay.Display(_username,st);

        //Process.Start($"notepad.exe", $"{DataFileManager.ResourceFolder+_username+".ElsaHistory"}");
    }
    
}